### ISSUE TITLE
Is `AssetServer::load_folder()` planned or even possible to be supported with `HttpWasmAssetReader`?

### ISSUE THREAD

**Issue by rosingrind** [bevyengine/bevy#10459]
It's about this error I've got while trying to run latest Bevy with WASM:

https://github.com/bevyengine/bevy/blob/0cc11791b9a55f5d310ee754c9f7254081e7138a/crates/bevy_asset/src/io/wasm.rs#L98

**Comment by johnny-smitherson** [comment]
The problem here is possibly there's no common format for listing folders over HTTP.

I use [nginx autoindex format = json](http://nginx.org/en/docs/http/ngx_http_autoindex_module.html) to get the asset list for each folder and load them all as asset untyped.

**Comment by nicopap** [comment]
Yeah. I think it doesn't make sense to implement this as part of bevy.

The correct way is define your own HTTP loader, with the bevy loader as a field. For all `AssetServer` methods already implemented, you would delegate to the bevy loader, and for `load_folder`, you would implement it yourself. For example, it could send a request to get a directory listing (typically a index.html) and then send request to load each file listed.

Since there is no standard on directory listing, there isn't a "one correct way" of implementing it, whatever we would chose to implement would satisfy only a minority of the bevy users.

**Comment by asafigan** [comment]
With asset preprocessing, Bevy could write a manifest file that the `HttpWasmAssetReader` could read to find the paths to all the files.
